Modules/_testcapimodule.c

```diff
@@ -3329,6 +3329,34 @@ test_critical_sections(PyObject *module, PyObject *Py_UNUSED(args))
     Py_RETURN_NONE;
 }
 
+// Used by `finalize_thread_hang`.
+#ifdef _POSIX_THREADS
+static void finalize_thread_hang_cleanup_callback(void *Py_UNUSED(arg)) {
+    // Should not reach here.
+    assert(0 && "pthread thread termination was triggered unexpectedly");
```

Comment:
I suggest to not rely on assertions which are removed in release build. Instead, use `Py_FatalError(...)`.